Give CMake project() info, and addressing change in LIBGIT2_SOVERSION of libgit2

blu-base opened the pull request:
CMake asked for the project() information, which is added in this PR. 
If this is desirable, this piece of information could be enriched with further info, as suggested in the documents of [cmake](https://cmake.org/cmake/help/latest/command/project.html)

Furthermore, since version 0.99 there is a minor but breaking change in the API of libgit2, see https://github.com/libgit2/libgit2/issues/5415#issuecomment-589614188

`LIBGIT2_SOVERSION` is now returning a string instead of an integer

This PR now uses `LIBGIT2_VER_MAJOR` and `LIBGIT2_VER_MINOR` to check for version 0.22 and above.

a17r committed:
Fix install basket.xml to XDG_MIME_INSTALL_DIR

Summary: Previous code did not work at all in a sandboxed packaging build.

Test Plan: Build and install succeeds.

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D27991

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

tosky committed:
Clean translations after the import

- remove the translation file, which have been copied to the
  KDE translators' repositories;
- create a more complete Messages.sh (top-level) and remove
  the old incomplete one, as well as extract-messages.sh;
- use ki18n_install to include the translations when available
  under po/.

pinotree committed:
fixuifiles

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

yurchor committed:
Fix various minor typos

CarlSchwan committed:
Modernize code 'modernize-use-override' with clang tidy

Test Plan: Still compile with less warning

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D27945

CarlSchwan committed:
Improve README

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D27946

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

CarlSchwan committed:
make links https in doc

yurchor committed:
Remove duplicate includes

CarlSchwan committed:
add .arcconfig

veggero committed:
Added support for libgit>=.99

Summary: LIBGIT2_SOVERSION no longer returns an integer in libgit>=.99; this adds support for that version and 1.0 to come.

Differential Revision: https://phabricator.kde.org/D27990

CarlSchwan committed:
Fix copyright header detected by ebn

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D27992

CarlSchwan committed:
Use char instead of string for efficency when possible

Summary: See http://ebn.kde.org/krazy/reports/playground/utils/basket/index.html

Test Plan: make still work

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D27993

veggero committed:
LIBGIT_VER_MINOR is supposed to be LIBGIT2_VER_MINOR

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D28001

CarlSchwan committed:
fix: deprecation QString::null

CarlSchwan committed:
Make C++ ctors that should be declared 'explicit' explicit

Summary:
Also replace some 0 with nullptr when appropriate

See http://ebn.kde.org/krazy/reports/playground/utils/basket/index.html

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D28004

CarlSchwan committed:
Use https instead of http and handle https url better

Test Plan: Compile

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D28005

yurchor committed:
Remove Qt 3 remnant

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

veggero committed:
Add copyright email

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D28335

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

CarlSchwan committed:
feat: update AboutData class

CarlSchwan committed:
refactor: remove dead code Application::newInstance()

veggero committed:
Prettify hover and selected notes

Summary: Gradients are not very 2020 :P

Test Plan:
Before:
{F8258497}
{F8258495}
After:
{F8260421}
{F8260419}

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D29167

CarlSchwan committed:
Remove Nepomukintegration

Summary: It is outdated and already disabled

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D29182

CarlSchwan committed:
Use SPDX copyright information

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D29183

CarlSchwan committed:
Change desktop file name

CarlSchwan committed:
Rename dbus interface

CarlSchwan committed:
change dbus interface name

CarlSchwan committed:
Add flatpak manifest

sowelipililimute committed:
Improved flatpak manifest

CarlSchwan committed:
Add clang format rules

veggero committed:
clang format

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D29193

CarlSchwan committed:
Update readme

CarlSchwan committed:
remove installer, we now have flatpak

CarlSchwan committed:
remove debian packaging stuff, this wasn't updated since 2010

CarlSchwan committed:
Add AppStream file

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D29195

veggero committed:
Modernize tags

Summary: Add new tags done by @manueljlin

Reviewers: ognarb

Reviewed By: ognarb

Subscribers: manueljlin

Differential Revision: https://phabricator.kde.org/D29196

CarlSchwan committed:
Use svg for tag icons

Test Plan:
```
rm -rf ~/.cache/icon-cache.kcache
and delete old icons in ~/.local/kde/share/icons/hicolor/16x16/actions/tag_*.png (not the checkbox)
```

Reviewers: niccolove

Reviewed By: niccolove

Subscribers: IlyaBizyaev

Differential Revision: https://phabricator.kde.org/D29205

CarlSchwan committed:
feat(licensing): Add license file for flapak manifest

CarlSchwan committed:
feat: Update .gitignore file

veggero committed:
Remove load animation

Summary: This patch removes the initial load animation

Reviewers: ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D29211

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

hellozee committed:
Switched SmallIcon with QIcon overloads

According to KIConThemes API docs SmallIcon was deprecated in favor of QICon with standard sizes.

Differential Revision: https://phabricator.kde.org/D29235

hellozee committed:
QFontMetrics::width() is deprecated

hellozee committed:
QPixmap::find(QString) was deprecated

hellozee committed:
Result is not being used

hellozee committed:
Someone forgot how QString::trimmed() works, :D

hellozee committed:
KIconLoader::DesktopIcon is also deprecated

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

ismailof committed:
Refactor BasekFactory::newBasket() to use deafult arguments

Summary: This simplifies a lot the calls to this function, since most just use the defaults.

Test Plan: It compiles and runs

Reviewers: ognarb, niccolove, alex

Reviewed By: ognarb, niccolove, alex

Differential Revision: https://phabricator.kde.org/D29267

ismailof committed:
Address some compiler warnings

Summary:
- Port from deprecated dark() and light()
- Port from deprecated DesktopIcon
- Use KIconLoader::StdSizes instead of harcoded sizes
- Use QString() for empty strings
- Minimal style changes around those

Test Plan:
Basket still compiles and works
Less compiler warnings

Reviewers: niccolove, ognarb

Reviewed By: ognarb

Subscribers: alex

Differential Revision: https://phabricator.kde.org/D29240

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

CarlSchwan committed:
Use clang tidy to modernize code (nullptr and override)

Reviewers: niccolove

Differential Revision: https://phabricator.kde.org/D29273

hellozee committed:
Added a .clang-format

CarlSchwan committed:
refactor: Color picker

Summary:
* Copy logic from QColorDialog
* Don't hide window
* Use new connect syntax
* use unique_ptr

Reviewers: iasensio, niccolove

Reviewed By: iasensio

Differential Revision: https://phabricator.kde.org/D29280

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

CarlSchwan committed:
Remove obsolte and broken importer

Summary:
* tuxkarts is dead since 2010
* KnowIt is from KDE3
* knotes an kjots importer were written before Akonadi so doesn't work on the current version of these softwares
* Sticky Notes was last updated 14 years ago
* tomboy doesn't exist anymore but there is tomboy-ng. maybe the file format is still compatible
* And I didn't test Jreepad but I don't think there are many Jreepad users

Test Plan: Compile and see the menu with less entries

Reviewers: niccolove

Reviewed By: niccolove

Subscribers: anthonyfieroni

Differential Revision: https://phabricator.kde.org/D29559

CarlSchwan committed:
refactor: remove obsolte animate args to some methods

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

ismailof committed:
Refactor: simplify NoteContent

Summary:
Small refactor to reduce the size of `NoteContent` class

Move `cssColorsName()` to `Tools`, and also use a simpler `QHash`
Move `loadNode()` to `NoteFactory`

Test Plan: It works just the same

Reviewers: alex, ognarb

Reviewed By: alex, ognarb

Differential Revision: https://phabricator.kde.org/D29686

CarlSchwan committed:
Refactor CMakeFiles

Summary:
* Use ECM for adding tests
* Use consistent naming for CMake functions
* Only add tests when adding -DBUILD_TESTING
* Code formatting
* Use option for options
* Use ECM for version number and change version number from 2.49.beta to 2.49.90

Test Plan: `make -j8 install` works and `make test` too

Reviewers: niccolove, iasensio, alex

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D29718

CarlSchwan committed:
feat: use directly file system to build flatpak

CarlSchwan committed:
Remove disabled Kontact integration

Summary: This part was never ported to Qt5 and it's not worth investing time to port it.

Test Plan: Compile and make test

Reviewers: niccolove

Reviewed By: niccolove

Differential Revision: https://phabricator.kde.org/D29730

TobiasFella committed:
Port away from QSignalMapper

Reviewers: niccolove, ognarb

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D29733

TobiasFella committed:
Remove redundant command line options

Summary: The --help and --version command line options are added automatically by KAboutData when adding the --author and --license options

Test Plan: No warnings about duplicate options && options still there

Reviewers: ognarb, niccolove

Reviewed By: ognarb

Differential Revision: https://phabricator.kde.org/D29734

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

CarlSchwan committed:
feat: Add image for gitlab repository

CarlSchwan committed:
feat: Update KAboutData and use gitlab url

TobiasFella committed:
Port to new signal/slot syntax

yurchor committed:
Fix minor typos

alex1701c committed:
Fix pixelated icons

CarlSchwan committed:
feat: Add gitlab-ci.yml

CarlSchwan committed:
fix: ci

TobiasFella committed:
Fix clang build by not exporting namespace

jbruechert committed:
Rename icon to match appstream ID

ismailof committed:
Simplify NoteContent types

CarlSchwan committed:
Feat: make it compile with QT_NO_KEYWORDS

CarlSchwan committed:
add cmake to mr

CarlSchwan committed:
Remove some clazy warnings

ismailof committed:
Simplify NoteContent tooltips

ismailof committed:
Small clean-up on BasketScene

ismailof committed:
Remove custom crash handler

ismailof committed:
Refactor out load/save methods from BasketScene

Extract load/save methods for raw files to a new namespace `FileStorage`

ismailof committed:
Replace custom DiskErrorDialog with ErrorMessage

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

ismailof committed:
Fix annoying crash on exit

Block emitting signals in `BasketScene` destructor.

The destructor calls `deleteNotes()`, which emits some signals to update the
status bar. This was crashing since `BNPview` was being already set to `nullptr`.

This seemed to me the simpler way to avoid this crash.

Commit:
GIT_SILENT made messages (after extraction)

yurchor committed:
Remove stale parts from docbook

yurchor committed:
Simplify documentation installation system

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

blu-base committed:
Fix virtual function call in constructors

clang analyzer indicated a couple of virtual func call in
constructors, mainly in subclasses of notecontent.

Most occurances of this problem are trival because the respective
subclasses currently aren't inherited by further subclasses.

blu-base committed:
Fix virtual function call in FileContent

the virtual linkLook() was called in constructor which is fixed
trivially.
However, SoundContent inherits FileContent in which SoundContent
intends to override the linkLook() function, but is not available
at during construction.
I had to copy the setFileName method to SoundContent and applied
the explicit function call. However, this is very inconvinent,
and disputable.

blu-base committed:
Fixing deprecated QFlags inizialization

Regarding the constructors of LinkLabels:
It is suggested not to initaliaze Qt::WindowFlags with a nullptr.
Instead the default constructor, Qt::WindowFlags() is to be used.
This is plugged into QFrame, there this constructor is already
called by default.

An equivalent change is made to empty Qt::KeyboardModifiers.

Commit:
Prevent possible Qt container detachment

clang analyzer detected a location where a QList element could be
detached. Following the example of KFileMetaData, qAsConst now
prevents detachment.

Reference:
https://invent.kde.org/frameworks/kfilemetadata/-/blob/master/tests/dump.cpp#L45

CarlSchwan committed:
Remove system tray and KPassivePopup

Summary:
Instead use KMessageWidget to indicate errors
And don't display informative message about the last user action

Test Plan: {F8317979}

Reviewers: niccolove, iasensio

Reviewed By: niccolove, iasensio

Differential Revision: https://phabricator.kde.org/D29732

Commit:
Migrate QPixmap::grabWindow()

Qt Doc suggests to replace it with QScreen::grabWindow.
To get the QScreen, this commit calls the primaryScreen property.

However, i could test wether this change impacts multi monitor
environments...

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

gl-bars commented:
We can just drop support for libgit2<0.22 (set minimum version in CMakeLists), then `#if`s can be removed.

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
Document BasKet archive file format

This commit adds some explanations to the archive format of
BasKet (.baskets), Such as the file header, and the contained
files.

CarlSchwan committed:
Port color picker to xdg-portal and add wayland support

Commit:
GIT_SILENT made messages (after extraction)

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
SVN_SILENT made messages (.desktop file) - always resolve ours

In case of conflict in i18n, keep the version of the branch "ours"
To resolve a particular conflict, "git checkout --ours path/to/file.desktop"

Commit:
GIT_SILENT made messages (after extraction)

Commit:
Search improvements:
 - Case insensitive search
 - Search within basket titles as well

Commit:
Local basket shortcuts fixed

Commit:
GIT_SILENT made messages (after extraction)

Commit:
Add clear formatting menu item for HTML notes

This allows clearing the formating for HTML notes. The confusing
negation in the title of Settings > Baskets "Keep text formatting.."
was also removed.

Commit:
Several clicks's necessary to close the Tags > Customize dialog - fixed.

How to reproduce:
 - open / close the "Tags > Customize" dialog several times
 - after that, it is necessarys to click Ok / Cancel button several times to close the dialog.

 Reason: Duplicate connections for the Tags menu to open the dialog were created.

Commit:
Icons restored:
- Tags icons in menus restored
- Default basket icon in the Baskets hierarchy tree list restored

Commit:
Root basket in the basket tree is not selected on startup - fixed

Commit:
Settings Dialog: very small window size - fixed

Solution: Settings Dialog expanded to cover the largest internal page

jbruechert committed:
Update flatpak runtime to 5.15

CarlSchwan committed:
cleanup unused argument

CarlSchwan committed:
remove dead variable

CarlSchwan committed:
remove dead variable

CarlSchwan committed:
remove dead variable
